Attachment node and processor cap settings (#148493)
elastic/elasticsearch: docs/reference/enrich-processor/attachment.md

The source field must be a base64 encoded binary. If you do not want to incur the overhead of converting back and forth between base64, you can use the CBOR format instead of JSON and specify the field as a bytes array instead of a string representation. The processor will skip the base64 decoding then.

> You can cap how large that raw source value may be using the processor option `max_field_bytes` and/or the node setting `ingest.attachment.max_field_size`. See [Limit raw attachment field size](#attachment-raw-field-size-limits).

Reviewer: ```suggestion
To limit the size of the raw source value, use the processor option `max_field_bytes` or the node setting `ingest.attachment.max_field_size`. For more details, refer to [Limit raw attachment field size](#attachment-raw-field-size-limits).
```

| `target_field` | no | attachment | The field that will hold the attachment information |
| `indexed_chars` | no | 100000 | The number of chars being used for extraction to prevent huge fields. Use `-1` for no limit. |
| `indexed_chars_field` | no | `null` | Field name from which you can overwrite the number of chars being used for extraction. See `indexed_chars`. |

> | `max_field_bytes` | no | `-1` | Maximum allowed size of the attachment `field` value in bytes: length of a string (if base64 in JSON, checked before base64 decoding) or `byte[]` length for binary (for example, CBOR). If set to `-1`, there is no per-processor limit. The node setting `ingest.attachment.max_field_size` also applies; refer to [Limit raw attachment field size](#attachment-raw-field-size-limits). |

Reviewer: ```suggestion
| `max_field_bytes` {applies_to}`stack: ga 9.5` | no | `-1` | Maximum allowed size of the attachment `field` value in bytes: length of a string (if base64 in JSON, checked before base64 decoding) or `byte[]` length for binary (for example, CBOR). If set to `-1`, there is no per-processor limit. The node setting `ingest.attachment.max_field_size` also applies; refer to [Limit raw attachment field size](#attachment-raw-field-size-limits). |
```

Reviewer: for more info on `{applies_to}`, see [Write cumulative documentation](https://www.elastic.co/docs/contribute-docs/how-to/cumulative-docs)

> | `max_field_bytes` | no | `-1` | Maximum allowed size of the attachment `field` value in bytes: length of a string (in case of base64 in JSON; checked before base64 decoding) or `byte[]` length for binary (for example CBOR). If `-1`, there is no per-processor limit. The node setting `ingest.attachment.max_field_size` is applied in addition; see [Limit raw attachment field size](#attachment-raw-field-size-limits). |

Reviewer: ```suggestion
| `max_field_bytes` | no | `-1` | Maximum allowed size of the attachment `field` value in bytes: length of a string (if base64 in JSON, checked before base64 decoding) or `byte[]` length for binary (for example, CBOR). If set to `-1`, there is no per-processor limit. The node setting `ingest.attachment.max_field_size` also applies; refer to [Limit raw attachment field size](#attachment-raw-field-size-limits). |
```

% TESTRESPONSE[s/"_seq_no": \d+/"_seq_no" : $body._seq_no/ s/"_primary_term" : 1/"_primary_term" : $body._primary_term/]

## Limit raw attachment field size [attachment-raw-field-size-limits]

> 

Reviewer: github keeps misrendering my suggestion because of all the backticks. this should be:

` ## Limit raw attachment field size [attachment-raw-field-size-limits]`
` ```{applies_to} `
` stack: ga 9.5 `
` ``` `

> In some cases, attachment sizes can result in significant memory overhead, as can any further processing (for example, by Tika). To restrict the raw size of the attachment source field, use the node setting `ingest.attachment.max_field_size` or the processor parameter `max_field_bytes`. Limits are enforced early in the process (for example, before base64 decoding) to avoid allocating or decoding oversized payloads. If a document exceeds the limit, use an `on_failure` handler to address the resulting ingest failure.

Reviewer: ```suggestion
In some cases, attachment sizes can result in significant memory overhead, as can any further processing (for example, by Apache Tika). To restrict the raw size of the attachment source field, use the node setting `ingest.attachment.max_field_size` or the processor parameter `max_field_bytes`. Limits are enforced early in the process (for example, before base64 decoding) to avoid allocating or decoding oversized payloads. If a document exceeds the limit, use an `on_failure` handler to address the resulting ingest failure.
```

> In certain cases, the sizes of the attachments may incur a significant memory overhead, including any further processing that might be required (e.g., by Tika). Either the node setting `ingest.attachment.max_field_size`, and/or the processor parameter `max_field_bytes`, can be used to restrict the raw size of the attachment source field. Limits are enforced early (e.g., before Base64 decoding) to avoid allocating or decoding oversized payloads. If a document exceeds the limit, the resulting ingest failure can be handled with an `on_failure` handler.

Reviewer: ```suggestion
In some cases, attachment sizes can result in significant memory overhead, as can any further processing (for example, by Tika). To restrict the raw size of the attachment source field, use the node setting `ingest.attachment.max_field_size` or the processor parameter `max_field_bytes`. Limits are enforced early in the process (for example, before base64 decoding) to avoid allocating or decoding oversized payloads. If a document exceeds the limit, use an `on_failure` handler to address the resulting ingest failure.
```

> The node setting has a default of `-1`, which means no node-wide cap. You can set either an absolute size using byte-size syntax (e.g., `10mb`, `1gb`), or a relative value as a percentage or ratio of the JVM's heap for the node (e.g., `5%` or `0.05`).

Reviewer: ```suggestion
The node setting defaults to `-1`, which means no node-wide limit. You can set either an absolute size using byte-size syntax (for example, `10mb`, `1gb`), or a relative value as a percentage or ratio of the JVM heap for the node (for example, `5%` or `0.05`).
```

> In contrast to the node setting, the `max_field_bytes` is a per attachment processor parameter which receives only integer values. The default is `-1`, meaning no per-processor cap (but note that the node setting, if set, can still apply). Note that the node setting is checked in addition to the per-processor setting.

Reviewer: ```suggestion
Unlike the node setting, `max_field_bytes` is an attachment processor parameter that accepts only integer values. The default is `-1`, meaning no per-processor limit, but the node setting still applies. 
```
